src/asyncIterableToBuffer.js

```diff
@@ -12,6 +12,11 @@ const handler = {
       console.log('New call to', key);
       obj._keys[key] = true;
     }
+    // if function, ensure context is present.
+    if(typeof obj[key] === 'function') {
```

Comment:
This block of code belongs to a Proxy handler implementation and it will proxy obj keys. It also includes functions which had their context lost on previous implementation.


This is something we definitely should revisit later, but for now I just would like to ensure functions has correct context.

Reply:
The problem is that the "this" context is no longer really correct overall.  The right solution is to add a custom proxy on any function call found.  Probably if typeof is function, and we didn't find one in the parent custom functions, then throw an exception with the name of the function.

Reply:
just to be clear there is no 2 different of "this". What will happen is: or "this" is bind to function call context or not. The actual implementation does not bind it. So, no matter if there is an overwritten of methods or the method comes from parent it the context will not be present.

If, even though, this is something you understand to better cover later I would be more than happy to revert my changes (as I said, it will throw an error for deflate studies, but we can cover later)

Reply:
Lets revert it for now, and then if you have a test case that causes the problem, then I'd like to go through it carefully.